cmd/restic/cmd_mount.go

```diff
@@ -160,6 +159,32 @@ func umount(mountpoint string) error {
 	return systemFuse.Unmount(mountpoint)
 }
 
+func checkMountPointValidity(mountpoint string) error {
+	// check if the mountpoint exist and if it's a directory
+	info, err := os.Stat(mountpoint)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return errors.Fatalf("the mount path doesn't exist: %q", mountpoint)
+		}
+		return errors.Fatalf("the mount path is not a directory or doesn't exist %q", mountpoint)
+	}
+	if !info.IsDir() {
+		return errors.Fatalf("the mount path is not a directory %q", mountpoint)
+	}
+	// create a temporary file to test permission whether it's writable directory.
+	tmpFile := filepath.Join(mountpoint, fmt.Sprintf("%d", time.Now().UnixNano()))
+	f, err := os.Create(tmpFile)
+	if err != nil {
+		// check with IsPermission(err) ?
+		return errors.Fatalf("the mount path is not writable directory: %q", mountpoint)
+	}
+	// fixme: check for error ?
+	f.Close()
+	os.Remove(tmpFile)
+
+	return nil
+}
```

Comment:
I don't see why any of this is necessary. When the mountpoint doesn't exist, bazil.org/fuse reports that:

    mountpoint does not exist: /tmp/foo

When the permission are wrong, it reports that:

    2020/10/11 16:21:43 mount helper error: fusermount: user has no write access to mountpoint /tmp/foo

Reply:
Checking about this earlier make it fail faster and remove unnecessary work and terminal output like opening repo, ...
For the end user, this is a more friendly error message that reading.

But I guess we can remove this commit and simply close the #1724, on my last test the message print without blocking.

Reply:
Let's assume for now that #1724 was caused by something external to restic. So please remove this commit for now.

Reply:
Hi, the commit was removed and the changelog updated.